Non-human primate adaptation of PostFreeSurfer.

mmilch01 opened the pull request:
For discussion. Please do not merge.

mmilch01 committed:
+NHP adaptation, initial commit. Not tested to work.

glasserm commented on PostFreeSurfer/PostFreeSurferPipeline.sh:
```diff
@@ -418,36 +442,40 @@ if ((doProcessing)); then
-if ((doQC)); then
+if (( doQC )); then
   log_Msg "Generating structural QC scene and snapshots"
+  #TODO: Do NHP need special QC?
```
We will find out if it breaks when we test.

glasserm commented on PostFreeSurfer/scripts/CreateMyelinMaps.sh:
```diff
@@ -179,6 +198,14 @@ if (( IsLongitudinal==0 )); then #In the longitudinal case, this functionality i
 	fslmaths "$OutputMNIT2wImage" -div "$BiasFieldOutput" "$OutputMNIT2wImageRestore"
 	fslmaths "$OutputMNIT2wImageRestore" -mas "$T1wMNIImageBrainMask" "$OutputMNIT2wImageRestoreBrain"
+
+	#TODO NHP: check if this is needed.
```
What does this do?

glasserm commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -174,6 +183,15 @@ for LowResMesh in ${LowResMeshes} ; do
+NonHumanSpecies=0
+if [ "$Species" != "human" ]; then NonHumanSpecies=1; fi
+
+#TODO NHP: Does this need to be conditioned by NonHumanSpecies?
```
We should always create this.

glasserm replied:
Just do it for both, as this is an improvement.

glasserm commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -231,6 +251,15 @@ if [ "$LongitudinalMode" != "TIMEPOINT_STAGE2" ]; then
       ${CARET7DIR}/wb_command -add-to-spec-file "$T1wFolder"/fsaverage_LR"$LowResMesh"k/"$Session"."$LowResMesh"k_fs_LR.wb.spec INVALID "$T1wFolder"/"$T1wImage".nii.gz
     done
 
+    #TODO: check if this is NHP or old HCP code.
```
Both should do this.

glasserm commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -447,6 +487,12 @@ for Hemisphere in L R ; do
         ${CARET7DIR}/wb_command -metric-math "ln(var) / ln (2)" "$AtlasSpaceFolder"/"$NativeFolder"/"$Session"."$Hemisphere".StrainR_FS.native.shape.gii -var var "$AtlasSpaceFolder"/"$NativeFolder"/"$Session"."$Hemisphere".StrainR_FS.native.shape.gii
         rm "$AtlasSpaceFolder"/"$NativeFolder"/"$Session"."$Hemisphere".Strain_FS.native.shape.gii
 
+        #TODO NHP: do we need this?
```
I would think this was the same across species.

glasserm commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -461,8 +507,45 @@ for Hemisphere in L R ; do
         if [ "$LongitudinalMode" == "NONE" ]; then
-            cp "$AtlasSpaceFolder"/"$NativeFolder"/"$Session"."$Hemisphere".sphere.rot.native.surf.gii "$AtlasSpaceFolder"/"$NativeFolder"/MSMSulc/${Hemisphere}.sphere_rot.surf.gii
-            $HCPPIPEDIR/global/scripts/MSMSulc.sh --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere"
+            #TODO: check NHP
```
I would think this was the same across species.

glasserm commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -599,10 +682,24 @@ for Hemisphere in L R ; do
+                ${CARET7DIR}/wb_command -add-to-spec-file "$T1wFolder"/fsaverage_LR"$HighResMesh"k/"$Subject"."$HighResMesh"k_fs_LR.wb.spec $Structure "$T1wFolder"/fsaverage_LR"$HighResMesh"k/"$Subject"."$Hemisphere"."$Surface"."$HighResMesh"k_fs_LR.surf.gii
+            done
+        fi
+        #TODO: NHP or old HCP?
```
Related to the above.

glasserm commented on PostFreeSurfer/PostFreeSurferPipeline.sh:
```diff
@@ -119,6 +121,15 @@ Longitudinal Freesurfer files for template are stored under <Subject>.long.<Temp
+#NHP options
+opts_AddOptional '--species' 'Species' 'Species to be processed' "Set target species for non-human primate processing [human]" "human"
+
+#TODO NHP: choose adequate default options
```
Need to invoke SetSpecies.sh in Batch file.

coalsont replied:
Note, `master` won't have `SetUpSPECIES.sh` until we merge that PR.  If we copy it to this PR now, we will probably have a merge conflict once we merge NHP PreFS.

mmilch01 replied:
> Let's make sure to use SetSpecies.sh as in PreFreeSurfer and handle any of the NHP specific parameters in there.

Yes, I tried duplicating the PreFreeSurfer logic in the said batch script. Is there any specific concern?

glasserm commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -231,6 +251,15 @@ if [ "$LongitudinalMode" != "TIMEPOINT_STAGE2" ]; then
+    #TODO: check if this is NHP or old HCP code.
+    if (( NonHumanSpecies )); then 
+        [ "${T2wImage}" != "NONE" ] && ${CARET7DIR}/wb_command -add-to-spec-file "$AtlasSpaceFolder"/fsaverage_LR"$LowResMesh"k/"$Subject"."$LowResMesh"k_fs_LR.wb.spec INVALID "$AtlasSpaceFolder"/"$AtlasSpaceT2wImage".nii.gz
+        ${CARET7DIR}/wb_command -add-to-spec-file "$AtlasSpaceFolder"/fsaverage_LR"$LowResMesh"k/"$Subject"."$LowResMesh"k_fs_LR.wb.spec INVALID "$AtlasSpaceFolder"/"$AtlasSpaceT1wImage".nii.gz
```
These two should be done elsewhere in the code.

mmilch01 commented on PostFreeSurfer/PostFreeSurferPipeline.sh:
```diff
@@ -119,6 +121,15 @@ Longitudinal Freesurfer files for template are stored under <Subject>.long.<Temp
+opts_AddOptional '--myelinmappingfwhm' 'MyelinMappingFWHM' 'float' "Myelin mapping FWHM" "2"
+opts_AddOptional '--surfacesmoothingfwhm' 'SurfaceSmoothingFWHM' "2"
+opts_AddOptional '--msmsulcconf' 'MSMSulcConf' 'string' 'MSMSulc configuration' "NONE"
+opts_AddOptional 'flatmaprootname' 'FlatMapRootName' 'string' 'Flat map root name' "NONE"
```
Ok.

mmilch01 commented on PostFreeSurfer/PostFreeSurferPipeline.sh:
```diff
@@ -129,6 +140,10 @@ fi
 #display the parsed/default values
 opts_ShowValues
 
+if [ "$LongitudinalMode" == "NONE" && "$Species" != "human" ]; then 
```
Ok

mmilch01 commented on PostFreeSurfer/PostFreeSurferPipeline.sh:
```diff
@@ -220,6 +235,11 @@ case "$ThicknessReg" in
 esac
 
+NonHumanSpecies=0
+if [ "$Species" != "human" ]; then 
```
Ok

Commit:
+addressing easier reviewer comments.

Commit:
+added longitudinal processing to MSMSulc

glasserm commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -461,8 +501,60 @@ for Hemisphere in L R ; do
+                #${CARET7DIR}/wb_command -metric-merge "$AtlasSpaceFolder"/"$NativeFolder"/"$Session"."$Hemisphere".StrainR_MSMSulc.native.shape.gii -metric "$AtlasSpaceFolder"/"$NativeFolder"/"$Session"."$Hemisphere".Strain_MSMSulc.native.shape.gii -column 2
+
+                #TODO NHP check if this is needed. No exact analog exists in code below.
+                ${CARET7DIR}/wb_command -metric-math "ln(var) / ln (2)" "$AtlasSpaceFolder"/"$NativeFolder"/"$Session"."$Hemisphere".StrainJ_MSMSulc.native.shape.gii -var var "$AtlasSpaceFolder"/"$NativeFolder"/"$Session"."$Hemisphere".StrainJ_MSMSulc.native.shape.gii
```
Comment out this group.

mmilch01 replied:
Done

glasserm commented on global/scripts/MSMSulc.sh:
```diff
@@ -22,6 +22,7 @@ opts_AddOptional '--msm-conf' 'ConfFile' 'conf file' "provide the name of the co
 opts_AddOptional '--hemi' 'Hemi' 'hemisphere' "provide hemisphere for registration, L=Left, R=Right, default B=Both" "B"
 opts_AddOptional '--refmesh' 'RefMesh' 'ref mesh' "provide alternate standard sphere, default 164k_fs_LR, use .HEMISPHERE. instead of .L. or .R."
 opts_AddOptional '--refdata' 'RefData' 'ref data' "provide alternate reference data, use .HEMISPHERE. instead of .L. or .R."
+opts_AddOptional '--species' 'Species' 'string'	"set species [Human]" "Human"
```
Remove if species not needed.

glasserm commented on global/scripts/MSMSulc.sh:
```diff
@@ -54,6 +55,9 @@ then
 	RefData="$SurfaceTemplateFolder"/HEMISPHERE.refsulc.164k_fs_LR.shape.gii
 fi
 
+NonHumanSpecies=0
```
Remove if species not needed.

mmilch01 replied:
Done.

Commit:
+added unsaved changes

glasserm commented on global/scripts/MSMSulc.sh:
```diff
@@ -82,6 +82,14 @@ for Hemisphere in $Hemi ; do
 	RefDataFile=$(basename -- "$RefData")
 	ReferenceData="$(cd "$(dirname -- "$RefData")"; pwd)/${RefDataFile/HEMISPHERE/$Hemisphere}"
+
+	if (( NonHumanSpecies )); then 
```
As we discussed this is already run.

coalsont commented on PostFreeSurfer/PostFreeSurferPipeline.sh:
```diff
@@ -129,6 +140,10 @@ fi
 #display the parsed/default values
 opts_ShowValues
 
+if [ "$LongitudinalMode" == "NONE" && "$Species" != "Human" ]; then 
```
`"$LongitudinalMode" == "NONE"` is probably the wrong condition to test here, currently this appears to trip when nonhuman is run non-longitudinally.

Commit:
+Example batch script for PostFreesurferNHP

Commit:
+address some reviewer comments to previous commit 2f5a420

Commit:
+Comments after comparing with MyelinMap_BC.sh, based on reviewer comments to 935d0ab

Commit:
Addressing https://github.com/Washington-University/HCPpipelines/pull/366#discussion_r2525007135

+comment out parameters set in SetUpSPECIES.sh
+correct parameter names

mmilch01 commented on PostFreeSurfer/PostFreeSurferPipeline.sh:
```diff
@@ -212,6 +223,10 @@ case "$LongitudinalMode" in
         ;;
 esac
 
+if [ "$Sesslist" != "NONE" && "$Species" != "Human" ]; then 
```
I changed this line to instead check the IsLongitudinal variable which is set above.

coalsont replied:
For readability, "not equal to false" is not as obvious as "equals true".  You can also combine math (implicit nonzero test) and string logic syntax with `&&` outside the parens/brackets:

```
if ((IsLongitudinal)) && [[ "$Species" != "Human" ]]
```

mmilch01 committed:
+address comment https://github.com/Washington-University/HCPpipelines/pull/366#discussion_r2525741257
+changes based on further discussion on CreateMyelinMaps.sh for commit efd433cda44a3904d6501932fc063ea6abf64a6d

glasserm commented on PostFreeSurfer/scripts/CreateMyelinMaps.sh:
```diff
@@ -264,139 +323,152 @@ for STRING in "$AtlasSpaceFolder"/"$NativeFolder"@native@roi "$AtlasSpaceFolder"
+		# TODO NHP seems to be NHP specific
+		${CARET7DIR}/wb_command -cifti-separate "$ReferenceMyelinMaps" COLUMN \
+			-metric CORTEX_LEFT "$SphereFolder"/"$Session".L.RefMyelinMap."$RefResMesh"k_fs_LR.func.gii \
+			-metric CORTEX_RIGHT "$SphereFolder"/"$Session".R.RefMyelinMap."$RefResMesh"k_fs_LR.func.gii
```
Yes but you need to provide the argument in your invocation of MyelinMap_BC when the Species is not Human.

glasserm commented on PostFreeSurfer/PostFreeSurferPipeline.sh:
```diff
@@ -350,10 +369,14 @@ if ((doProcessing)); then
-    argList+=("$SessionList")  # ${27}
+    argList+=("$SessionList")               # ${27}
+    argList+=("$Species")                   # ${28} NHP parameters
+    argList+=("MSMSulcConf")                # ${29}
```
Not a variable.

glasserm commented on PostFreeSurfer/PostFreeSurferPipeline.sh:
```diff
@@ -350,10 +369,14 @@ if ((doProcessing)); then
+    argList+=("$SessionList")               # ${27}
+    argList+=("$Species")                   # ${28} NHP parameters
+    argList+=("MSMSulcConf")                # ${29}
+    argList+=("FlatMapRootName")            # #{30} 
```
Not a variable.

Commit:
+Changes addressing reviewer comments:
https://github.com/Washington-University/HCPpipelines/pull/366#discussion_r2582590447
https://github.com/Washington-University/HCPpipelines/pull/366#discussion_r2578833742
https://github.com/Washington-University/HCPpipelines/pull/366#discussion_r2582593995
https://github.com/Washington-University/HCPpipelines/pull/366#discussion_r2582594326
https://github.com/Washington-University/HCPpipelines/pull/366#discussion_r2582594782
https://github.com/Washington-University/HCPpipelines/pull/366#discussion_r2582595206

glasserm commented on Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh:
```diff
@@ -0,0 +1,171 @@
+    #LowResMeshes="32" #Usually 32k vertices, if multiple delimit with @, must already exist in templates dir
+    SubcorticalGrayLabels="${HCPPIPEDIR_Config}/FreeSurferSubcorticalLabelTableLut.txt"
+    FreeSurferLabels="${HCPPIPEDIR_Config}/FreeSurferAllLut.txt"
+    ReferenceMyelinMaps="${HCPPIPEDIR_Templates}/standard_mesh_atlases/Conte69.MyelinMap_BC.164k_fs_LR.dscalar.nii"
```
```suggestion
    #ReferenceMyelinMaps="${HCPPIPEDIR_Templates}/standard_mesh_atlases/Conte69.MyelinMap_BC.164k_fs_LR.dscalar.nii"
```

glasserm commented on Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh:
```diff
@@ -0,0 +1,171 @@
+        --refmyelinmaps=$ReferenceMyelinMaps \
+        --regname=$RegName \
+        --use-ind-mean="$UseIndMean"" \
+        --species="$Species"
```
Missing arguments from above.

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

Commit:
+addressing reviewer comments
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2795602065
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2795603039

coalsont commented on Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh:
```diff
@@ -0,0 +1,180 @@
+        --use-ind-mean="$UseIndMean"\
+        --species="$Species"
+        --mcsigma="$CorrectionSigma"
+        --myelin-voume-fwhm="$MyelinMappingFWHM"
```
While line continuations aren't needed in the context of `...=()`, at least be consistent:
```suggestion
        --use-ind-mean="$UseIndMean" \
        --species="$Species" \
        --mcsigma="$CorrectionSigma" \
        --myelin-voume-fwhm="$MyelinMappingFWHM" \
```

coalsont commented on PostFreeSurfer/scripts/CreateMyelinMaps.sh:
```diff
@@ -223,6 +257,29 @@ for Hemisphere in L R ; do
     rm "$T1wFolder"/temp_ribbon.nii.gz
     ${CARET7DIR}/wb_command -metric-smoothing "$T1wFolder"/"$NativeFolder"/"$Session"."$Hemisphere".midthickness.native.surf.gii "$AtlasSpaceFolder"/"$NativeFolder"/"$Session"."$Hemisphere".MyelinMap.native.func.gii "$SurfaceSmoothingSigma" "$AtlasSpaceFolder"/"$NativeFolder"/"$Session"."$Hemisphere".SmoothedMyelinMap.native.func.gii -roi "$AtlasSpaceFolder"/"$NativeFolder"/"$Session"."$Hemispher ...
+
+	#TODO: Check NHP or old HCP?
```
What is this and all the below TODO comments?

mmilch01 replied:
Addressed in b2dda3d0fc6e7c564f5293430adc0b7c829097fa

glasserm commented on Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh:
```diff
@@ -0,0 +1,180 @@
+get_batch_options "$@"
+
+StudyFolder="${HOME}/projects/HCPPipelines_ExampleData" #Location of Session folders (named by subjectID)
+Sessionlist="100307 100610" #Space delimited list of subject IDs
```
```suggestion
Subjlist="100307 100610" #Space delimited list of subject IDs
```

glasserm commented on Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh:
```diff
@@ -0,0 +1,180 @@
+######################################### DO WORK ##########################################
+
+
+for Session in $Sesslist ; do
```
```suggestion
for Subject in $Subjlist ; do
```

glasserm commented on Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh:
```diff
@@ -0,0 +1,180 @@
+
+
+for Session in $Sesslist ; do
+    echo $Session
```
```suggestion
    echo $Subject
```

glasserm commented on Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh:
```diff
@@ -0,0 +1,180 @@
+    fi
+    job=("${queuing_command[@]}" "$HCPPIPEDIR"/PostFreeSurfer/PostFreeSurferPipeline.sh \
+        --study-folder="$StudyFolder" \
+        --session="$Session" \
```
```suggestion
        --subject="$Subject" \
```

mmilch01 committed:
Update PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh

Co-authored-by: Tim Coalson <coalsont@users.noreply.github.com>

mmilch01 committed:
Update Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Update Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Update Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh

Co-authored-by: Tim Coalson <coalsont@users.noreply.github.com>

mmilch01 committed:
Update Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Update Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Update Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Update Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Update Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Update Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Update Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Update PostFreeSurfer/scripts/CreateMyelinMaps.sh

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

Commit:
+Cleanup. Addressing comments:
https://github.com/Washington-University/HCPpipelines/pull/366#discussion_r2843722903
https://github.com/Washington-University/HCPpipelines/pull/366#discussion_r2855685912
https://github.com/Washington-University/HCPpipelines/pull/366#discussion_r2855692384

mmilch01 commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -461,8 +501,60 @@ for Hemisphere in L R ; do
+            #TODO: check NHP
+            if (( NonHumanSpecies )); then 
+
+                #the following three lines are moved to MSMSulc.sh
```
Addressed in b2dda3d0fc6e7c564f5293430adc0b7c829097fa

Commit:
+unsaved commit

glasserm commented on Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh:
```diff
@@ -0,0 +1,164 @@
+get_batch_options "$@"
+
+StudyFolder="${HOME}/projects/HCPPipelines_ExampleData" #Location of Session folders (named by subjectID)
+Subjlist="100307 100610" #Space delimited list of subject IDs
```
```suggestion
Subjlist="Macaque1 Macaque2" #Space delimited list of subject IDs
```

mmilch01 committed:
Update Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Update Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Update Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Update Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
+finish cleanup of TODO comments

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

glasserm commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -127,6 +127,15 @@ log_Msg "InflateExtraScale: ${InflateExtraScale}"
+Log_Msg "Species: $Species"
+
+MSMSulcConf="${26}"
+Log_Msg "MSMSulcConf: $MSMSulcConf"
```
```suggestion
log_Msg "MSMSulcConf: $MSMSulcConf"
```

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

glasserm commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -194,11 +209,14 @@ if [ "$LongitudinalMode" != "TIMEPOINT_STAGE2" ]; then
+            mri_convert -rt nearest -rl "$T1wFolder"/"$T1wImage".nii.gz "$FreeSurferFolder"/mri/"$Image".mgz "$T1wFolder"/"$Image"_1mm.nii.gz
+            if (( NonHumanSpecies )); then 
+                applywarp --rel --interp=nn -i "$T1wFolder"/"$T1wImageBrainMask"_1mm.nii.gz -r "$AtlasSpaceFolder"/"$AtlasSpaceT1wImage" --premat=$FSLDIR/etc/flirtsch/ident.mat -o "$T1wFolder"/"$T1wImageBrainMask".nii.gz
+            fi
```
```suggestion
```

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

glasserm commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -422,11 +450,13 @@ for Hemisphere in L R ; do
+        elif [ -e "$SurfaceAtlasDIR"/colin.cerebral."$Hemisphere".flat."$HighResMesh"k_fs_LR.surf.gii ] ; then
+                cp "$SurfaceAtlasDIR"/colin.cerebral."$Hemisphere".flat."$HighResMesh"k_fs_LR.surf.gii "$AtlasSpaceFolder"/"$Session"."$Hemisphere".flat."$HighResMesh"k_fs_LR.surf.gii
+        fi
+        ${CARET7DIR}/wb_command -add-to-spec-file "$AtlasSpaceFolder"/"$Session"."$HighResMesh"k_fs_LR.wb.spec $Structure "$AtlasSpaceFolder"/"$Session"."$Hemisphere".flat."$HighResMesh"k_fs_LR.surf.gii
```
Also this block is not executing the copy at all despite the file being in the expected location.

glasserm commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -422,11 +450,13 @@ for Hemisphere in L R ; do
-            ${CARET7DIR}/wb_command -add-to-spec-file "$AtlasSpaceFolder"/"$Session"."$HighResMesh"k_fs_LR.wb.spec $Structure "$AtlasSpaceFolder"/"$Session"."$Hemisphere".flat."$HighResMesh"k_fs_LR.surf.gii
-        fi
 
+        if (( NonHumanSpecies )) && [ -e "$SurfaceAtlasDIR"/${FlatMapRootName}."$Hemisphere".flat."$HighResMesh"k_fs_LR.surf.gii ] ; then
```
```suggestion
        if [[ (( NonHumanSpecies )) && -e "$SurfaceAtlasDIR"/${FlatMapRootName}."$Hemisphere".flat."$HighResMesh"k_fs_LR.surf.gii ]] ; then
```

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

glasserm commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -422,11 +450,19 @@ for Hemisphere in L R ; do
-            ${CARET7DIR}/wb_command -add-to-spec-file "$AtlasSpaceFolder"/"$Session"."$HighResMesh"k_fs_LR.wb.spec $Structure "$AtlasSpaceFolder"/"$Session"."$Hemisphere".flat."$HighResMesh"k_fs_LR.surf.gii
-        fi
 
+        if [ (( NonHumanSpecies )) ] ; then
```
```suggestion
        if (( NonHumanSpecies )) ; then
```

mmilch01 committed:
Apply suggestion from @glasserm

Co-authored-by: glasserm <glasserm@wusm.wustl.edu>

glasserm commented on PostFreeSurfer/PostFreeSurferPipeline.sh:
```diff
@@ -418,36 +441,40 @@ if ((doProcessing)); then
+    argList+=("$ThicknessReg")                              # ${42}
+    argList+=("$Species")                                   # ${43} NHP parameters
+    argList+=("$MyelinVolumeFWHM")                          # ${44}
+    argList+=("$MyelinSurfaceFWHM")                       # ${45}
```
Probably should convert these too.

Commit:
+replaced positional arguments in FreeSurfer2CaretConvertAndRegisterNonlinear.sh, CreateMyelinMaps.sh
Addressing these comments:
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2927878735
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2927878735

mmilch01 commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -127,6 +127,15 @@ log_Msg "InflateExtraScale: ${InflateExtraScale}"
+
+FlatMapRootName="${27}"
+log_Msg "FlatMapRootName: $FlatMapRootName"
+
```
Addressed here: e07c7ee423585df77bb1b9db6442a86202cfc9a8

mmilch01 committed:
+fix opts_addOptional function name

mmilch01 committed:
+remove --species argument that is no longer required and breaks processing

mmilch01 committed:
+added missing MSMConfig files

mmilch01 committed:
+fix array assignment

mmilch01 committed:
+added --surface-atlas-dir to CreateMyelinMaps.sh, and replace hard-coded value for humans on line 316
to address comment:
https://github.com/Washington-University/HCPpipelines/pull/366#issuecomment-4070621653

mmilch01 committed:
+clean up.

mmilch01 committed:
+clean up

coalsont commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -29,15 +15,8 @@ if [ -z "${HCPPIPEDIR}" ]; then
 fi
 
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
 source "${HCPPIPEDIR}/global/scripts/debug.shlib" "$@"         # Debugging functions; also sources log.shlib
```
Since this script was already using `debug.shlib`, and we are testing it anyway, we should probably `set -eu` it (at the top, like the newopts example).

coalsont commented on PostFreeSurfer/scripts/CreateMyelinMaps.sh:
```diff
@@ -47,54 +26,79 @@ log_Check_Env_Var HCPPIPEDIR
+opts_AddMandatory '--surface-atlas-dir' 'SurfaceAtlasDIR' 'path' "surface atlas directory"
+
+opts_ParseArguments "$@"
+
```
Use the whole newopts template, every time:

https://github.com/Washington-University/HCPpipelines/blob/0d21f09faeab5fbf3d9574bb6e6b84c86495d5e7/global/scripts/newopts.shlib#L35-L38

mmilch01 committed:
Update PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh

Co-authored-by: Tim Coalson <coalsont@users.noreply.github.com>

mmilch01 committed:
Update PostFreeSurfer/scripts/CreateMyelinMaps.sh

Co-authored-by: Tim Coalson <coalsont@users.noreply.github.com>

mmilch01 committed:
+address the comments:
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2968175279
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2968155361
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2968182582
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2968169990
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2968159122
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2968168152
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2968127686
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2968185129
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2968131099
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2968189751

mmilch01 commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -29,15 +15,8 @@ if [ -z "${HCPPIPEDIR}" ]; then
   exit 1
```
Addressed in caa960849921bf9b2c7853ac5a3f78ecdb348dee

mmilch01 commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -134,11 +95,12 @@ LongitudinalMode="${24}"
 #LIST of all timepoints, @ separated
-LongitudinalTimepoints="${27}"
+#LongitudinalTimepoints="${27}"
+
```
Addressed in caa960849921bf9b2c7853ac5a3f78ecdb348dee

mmilch01 commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -599,10 +640,19 @@ for Hemisphere in L R ; do
+            ${CARET7DIR}/wb_command -surface-resample "$T1wFolder"/"$NativeFolder"/"$Session"."$Hemisphere"."$Surface".native.surf.gii ${RegSphere} "$AtlasSpaceFolder"/"$Session"."$Hemisphere".sphere."$HighResMesh"k_fs_LR.surf.gii BARYCENTRIC "$T1wFolder"/fsaverage_LR"$HighResMesh"k/"$Session"."$Hemisphere"."$Surface"."$HighResMesh"k_fs_LR.surf.gii
+            ${CARET7DIR}/wb_command -add-to-spec-file "$T1wFolder"/fsaverage_LR"$HighResMesh"k/"$Session"."$HighResMesh"k_fs_LR.wb.spec $Structure "$T1wFolder"/fsaverage_LR"$HighResMesh"k/"$Session"."$Hemisphere"."$Surface"."$HighResMesh"k_fs_LR.surf.gii
+        done
+        #TODO: new feature from NHP, pay attention in tests.
```
Same as above, TODOs removed.

coalsont commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -1,43 +1,20 @@
+then 
+    pipedirguessed=1 
+    #fix this if the script is more than one level below HCPPIPEDIR 
+    export HCPPIPEDIR="$(dirname -- "$0")/.." 
```
```suggestion
    export HCPPIPEDIR="$(dirname -- "$0")/../.." 
```

mmilch01 committed:
Update PostFreeSurfer/scripts/CreateMyelinMaps.sh

Co-authored-by: Tim Coalson <coalsont@users.noreply.github.com>

mmilch01 committed:
Update PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh

Co-authored-by: Tim Coalson <coalsont@users.noreply.github.com>

Commit:
+address:
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2984513258
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2984478119
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2984478979
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2985246456
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r2985218564

mmilch01 committed:
Apply suggestion from @coalsont

Co-authored-by: Tim Coalson <coalsont@users.noreply.github.com>

coalsont commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -49,102 +26,67 @@ log_Check_Env_Var CARET7DIR
+opts_AddMandatory '--freesurfer-folder' 'FreeSurferFolder' 'folder' "FreeSurfer folder"
+opts_AddOptional '--freesurfer-input' 'FreeSurferInput' "FreeSurfer input image" "T1w_acpc_dc_restore_1mm"
+opts_AddOptional '--t1w-image' 'T1wImage' "T1w image" "T1w_acpc_dc"
+opts_AddOptional '--t2w-image' 'T2wImage' "T2w image" "T2w_acpc_dc"
```
If these were intended to be defaults, they will be used as the description instead, as currently written (you need 1 more argument to `opts_AddOptional`, the "typename-ish" string).

However, what PostFS actually passes to this script for t1w and t2w is:

```
T1wRestoreImage="T1w_acpc_dc_restore"
T2wRestoreImage="T2w_acpc_dc_restore"
...
        --t1w-image="$T1wRestoreImage"
        --t2w-image="$T2wRestoreImage"
```

mmilch01 replied:
Fixed in f7fe65bdb0e15b592618ba019b700d9039a6ae1b

coalsont commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -49,102 +26,67 @@ log_Check_Env_Var CARET7DIR
+opts_AddMandatory '--inverse-atlas-transform' 'InverseAtlasTransform' 'path' "inverse atlas transform"
+opts_AddOptional '--atlas-space-t1w-image' 'AtlasSpaceT1wImage' "atlas space T1w image" "T1w_restore"
+opts_AddOptional '--atlas-space-t2w-image' 'AtlasSpaceT2wImage' "atlas space T2w image" "T2w_restore"
+opts_AddOptional '--t1w-image-brain-mask' 'T1wImageBrainMask' "T1w image brain mask" "brainmask_fs"
```
Similarly, these have what looks like a default value in the argument for their description.

mmilch01 replied:
Fixed in f7fe65bdb0e15b592618ba019b700d9039a6ae1b

mmilch01 committed:
Update PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh

Co-authored-by: Tim Coalson <coalsont@users.noreply.github.com>

mmilch01 committed:
Apply suggestions from code review

Co-authored-by: Tim Coalson <coalsont@users.noreply.github.com>

mmilch01 committed:
Address reviewer comments:
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r3025131277
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r3025144037
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r3025156461
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r3025024070
https://github.com/Washington-University/HCPpipelines/pull/366/changes#r3025048201

mmilch01 commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -462,7 +432,12 @@ for Hemisphere in L R ; do
+                $HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf=${MSMCONFIGDIR}/${MSMSulcConf} --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere"
+            else                 
+                $HCPPIPEDIR/global/scripts/MSMSulc.sh --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere"
+            fi
```
Made the NHP line always execute in f7fe65bdb0e15b592618ba019b700d9039a6ae1b

mmilch01 commented on PostFreeSurfer/PostFreeSurferPipeline.sh:
```diff
@@ -88,6 +88,8 @@ opts_AddOptional '--mcsigma' 'CorrectionSigma' 'number' "myelin map bias correct
 opts_AddOptional '--inflatescale' 'InflateExtraScale' 'number' "surface inflation scaling factor to deal with different resolutions, default '1'" '1'
 opts_AddOptional '--processing-mode' 'ProcessingMode' 'HCPStyleData|LegacyStyleData' "disable some HCP preprocessing requirements to allow processing of data that doesn't meet HCP acquisition guidelines - don't use this if you don't need to" 'HCPStyleData'
+
+
 opts_AddOptional '--structural-qc' 'QCMode' 'yes|no|only' "whether to run structural QC, default 'yes'" 'yes'
```
I don't see a reason either. Removed these spaces in f7fe65bdb0e15b592618ba019b700d9039a6ae1b

mmilch01 commented on PostFreeSurfer/PostFreeSurferPipeline.sh:
```diff
@@ -119,6 +121,14 @@ Longitudinal Freesurfer files for template are stored under <Subject>.long.<Temp
+opts_AddOptional '--species' 'Species' 'Species to be processed' "Set target species for non-human primate processing [Human]" "Human"
+
+opts_AddOptional '--myelin-volume-fwhm' 'MyelinVolumeFWHM' 'float' "Myelin mapping volume smoothing FWHM" "2"
+opts_AddOptional '--myelin-surface-fwhm' 'MyelinSurfaceFWHM' 'float' "Myelin mapping surface smoothing FWHM" "2"
```
Changed defaults in f7fe65bdb0e15b592618ba019b700d9039a6ae1b.

coalsont commented on PostFreeSurfer/PostFreeSurferPipeline.sh:
```diff
@@ -119,6 +119,14 @@ Longitudinal Freesurfer files for template are stored under <Subject>.long.<Temp
+
+opts_AddOptional '--myelin-volume-fwhm' 'MyelinVolumeFWHM' 'float' "Myelin mapping volume smoothing FWHM" "5"
+opts_AddOptional '--myelin-surface-fwhm' 'MyelinSurfaceFWHM' 'float' "Myelin mapping surface smoothing FWHM" "4"
+opts_AddOptional '--msmsulc-conf' 'MSMSulcConf' 'string' 'MSMSulc configuration' "MSMSulcStrainFinalconf"
```
`MSMSulc.sh` (currently) appears to expect a complete path to the config, but this default name is relative to the `$MSMCONFIGDIR` (and similarly, the settings in `SetUpSPECIES.sh` don't have a path).  I don't know if `msm` reads the `$MSMCONFIGDIR` environment variable.  ~Assuming it doesn't, the simple thing to do is to handle it like `MSMSulc.sh` does internally:~

`<suggestion removed>`

Alternatively, somewhere (maybe in `MSMSulc.sh` itself) could check whether the value contains a `/`, and if not, prepend the `$MSMCONFIGDIR`.

Edit: since this is exposed to users and currently set without the path in `SetUpSPECIES.sh`, adding this functionality to `MSMSulc.sh` is a better idea (again, assuming `msm` itself doesn't know about `MSMCONFIGDIR`) - in particular, I'd prefer to avoid making the order of sourcing `SetUpHCPPipelines.sh` versus `SetUpSPECIES.sh` important.

mmilch01 replied:
Handling the conf file path resolution in MSMSulc itself is a potential source of error when the calling pipeline (e.g. PostFS) has progressed substantially but MSMCONFIGDIR wasn't defined. 
The only place in HCP pipelines where MSMSulc.sh is called from, is the PostFreeSurfer pipeline. Any reason why MSMSulc.sh shouldn't always require the full path in --msm-conf? This would also force to put the conf path resolution in the beginning of the calling pipeline and help readability. Legacy code shouldn't be affected, since we'll change both PostFreeSurferPipeline.sh and MSMSulc.sh simultaneously

coalsont replied:
This env variable has been somewhere in SetUpHCP... since 2017:

https://github.com/Washington-University/HCPpipelines/commit/962f018e649a0e90e249f6d215827e6971308eed

If we are worried that someone has a really old version of SetUpHCP... and they get a delayed error (though PostFS was relatively fast), we can check that it is set at the beginning of PostFS.  Requiring absolute path (when we have an env variable that appears to be intended for this purpose) seems like a trap for users and developers (and requires some contortions in the current code).

mmilch01 replied:
addressed in 68235fb68481c35fa4338820cf0bdac46651e742

coalsont replied:
See my additional comments implementing the "contains a `/`" logic I suggested.

Commit:
+address reviewer comment https://github.com/Washington-University/HCPpipelines/pull/366/changes#r3034225911

coalsont commented on PostFreeSurfer/PostFreeSurferPipeline.sh:
```diff
@@ -129,6 +137,10 @@ fi
 #display the parsed/default values
 opts_ShowValues
 
+if [ ! -d "$MSMCONFIGDIR" ]; then 
```
Check for unset (or `""`), rather than folder existing (if you want to test `-d`, give it a separate error message).  For convenience, also don't error if the config file already has a path:

```suggestion
if [[ "$MSMSulcConf" != */* && "${MSMCONFIGDIR:-}" == "" ]]; then 
```

coalsont commented on global/scripts/MSMSulc.sh:
```diff
@@ -38,6 +38,14 @@ SurfaceTemplateFolder="$HCPPIPEDIR"/global/templates/standard_mesh_atlases
 NativeFolder="$NonlinearFolder"/Native
 
+#Try to append $MSMCONFIGDIR to ConfFile if the file does not exist.
+if [ ! -f "$ConfFile" ] && [ -d "${MSMCONFIGDIR}" ]; then 
```
Let's make this behavior predictable without testing the filesystem:

```suggestion
if [[ "$ConfFile" != */* ]]; then 
```

We don't need to check anything about `$MSMCONFIGDIR` here at all - if it is unset, `set -eu` will error and explain that, and if it is wrong, the wrong path will be in the `$ConfFile` variable that is already in the next error check/message.

coalsont commented on global/scripts/MSMSulc.sh:
```diff
@@ -38,6 +38,14 @@ SurfaceTemplateFolder="$HCPPIPEDIR"/global/templates/standard_mesh_atlases
 NonlinearFolder="$SubjectDir"/"$Subject"/MNINonLinear
 NativeFolder="$NonlinearFolder"/Native
 
+#Try to append $MSMCONFIGDIR to ConfFile if the file does not exist.
```
Update comment to match my suggested logic, too:

```suggestion
#If the config file doesn't have a directory separator, treat it as relative to $MSMCONFIGDIR
```

mmilch01 committed:
Apply suggestions from code review

Co-authored-by: Tim Coalson <coalsont@users.noreply.github.com>

mmilch01 committed:
+remove unsupported argument

mmilch01 committed:
+Remove references to unbound script_name variable

AutioJ commented on Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh:
```diff
@@ -0,0 +1,166 @@
+#  environment: HCPPIPEDIR, FSLDIR, CARET7DIR
+
+#Set up pipeline environment variables and software
+source "$EnvironmentScript"
```
Should be before SetUpSPECIES.sh in L69

mmilch01 replied:
> Should be before SetUpSPECIES.sh in L69

Addressed in 5d79fe1dd5ade4870d0640a27413acafcefaae57

mmilch01 committed:
+address comment:
https://github.com/Washington-University/HCPpipelines/pull/366#discussion_r3169404491

glasserm commented on Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh:
```diff
@@ -0,0 +1,165 @@
+# LowResMeshes="32@20" #Needs to match what is in PostFreeSurfer
+# FinalfMRIResolution="1.6" #Needs to match what is in fMRIVolume
+# SmoothingFWHM="1.6" #Recommended to be roughly the voxel size
+# GrayordinatesResolution="1.6" #should be either 1 (7T) or 2 (3T) for human.
```
Use MacaqueRhesus values.

mmilch01 replied:
Addressed in 88c808014edfd06cf62fde7e7cd95a82b7e682d6

glasserm commented on Examples/Scripts/PostFreeSurferPipelineBatchNHP.sh:
```diff
@@ -0,0 +1,165 @@
+
+    SubcorticalGrayLabels="${HCPPIPEDIR_Config}/FreeSurferSubcorticalLabelTableLut.txt"
+    FreeSurferLabels="${HCPPIPEDIR_Config}/FreeSurferAllLut.txt"
+    #ReferenceMyelinMaps="${HCPPIPEDIR_Templates}/standard_mesh_atlases/Conte69.MyelinMap_BC.164k_fs_LR.dscalar.nii"
```
Use MacaqueRhesus values.

mmilch01 replied:
Addressed in 88c808014edfd06cf62fde7e7cd95a82b7e682d6

glasserm commented:
Line 435: `$HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf=${MSMCONFIGDIR}/${MSMSulcConf} --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session".HEMISPHERE.sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/${Session}.HEMISPHERE.refsulc."$HighResMesh"k_fs_LR.shape.gii`

Line 448: `$HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf=${MSMCONFIGDIR}/${MSMSulcConf} --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session".HEMISPHERE.sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/${Session}.HEMISPHERE.refsulc."$HighResMesh"k_fs_LR.shape.gii`

_TSC EDIT: refmesh/refsulc expect placeholders, as MSMSulc.sh can loop over hemispheres_

mmilch01 committed:
+address comments:
https://github.com/Washington-University/HCPpipelines/pull/366#discussion_r3184518938
https://github.com/Washington-University/HCPpipelines/pull/366#discussion_r3184521759
https://github.com/Washington-University/HCPpipelines/pull/366#issuecomment-4374805232

mmilch01 commented:
> Line 435: `$HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf=${MSMCONFIGDIR}/${MSMSulcConf} --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session"."$Hemisphere".sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/${Session}.${Hemisphere}.refsulc."$HighResMesh"k_fs_LR.shape.gii`
> 
> Line 448: `$HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf=${MSMCONFIGDIR}/${MSMSulcConf} --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session"."$Hemisphere".sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/${Session}.${Hemisphere}.refsulc."$HighResMesh"k_fs_LR.shape.gii`

Addressed in 88c808014edfd06cf62fde7e7cd95a82b7e682d6

coalsont commented:
> > Line 435: `$HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf=${MSMCONFIGDIR}/${MSMSulcConf} --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session"."$Hemisphere".sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/${Session}.${Hemisphere}.refsulc."$HighResMesh"k_fs_LR.shape.gii`
> > Line 448: `$HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf=${MSMCONFIGDIR}/${MSMSulcConf} --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session"."$Hemisphere".sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/${Session}.${Hemisphere}.refsulc."$HighResMesh"k_fs_LR.shape.gii`
> 
> Addressed in [88c8080](https://github.com/Washington-University/HCPpipelines/commit/88c808014edfd06cf62fde7e7cd95a82b7e682d6)

The current commit doesn't have --refmesh/--refsulc, which default to the human templates in `HCPPIPEDIR`, not the copies in the subject.  Matt's suggestion is to point `MSMSulc.sh` to the copies made earlier in the script.

glasserm commented:
> > > Line 435: `$HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf=${MSMCONFIGDIR}/${MSMSulcConf} --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session"."$Hemisphere".sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/${Session}.${Hemisphere}.refsulc."$HighResMesh"k_fs_LR.shape.gii`
> > > Line 448: `$HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf=${MSMCONFIGDIR}/${MSMSulcConf} --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session"."$Hemisphere".sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/${Session}.${Hemisphere}.refsulc."$HighResMesh"k_fs_LR.shape.gii`
> > 
> > 
> > Addressed in [88c8080](https://github.com/Washington-University/HCPpipelines/commit/88c808014edfd06cf62fde7e7cd95a82b7e682d6)
> 
> The current commit doesn't have --refmesh/--refsulc, which default to the human templates in `HCPPIPEDIR`, not the copies in the subject. Matt's suggestion is to point `MSMSulc.sh` to the copies made earlier in the script.

@mmilch01 you didn't actually fix the bug!

mmilch01 committed:
+address https://github.com/Washington-University/HCPpipelines/pull/366#issuecomment-4392846131

mmilch01 commented:
> $HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf=${MSMCONFIGDIR}/${MSMSulcConf} --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session".HEMISPHERE.sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/${Session}.HEMISPHERE.refsulc."$HighResMesh"k_fs_LR.shape.gii

Sorry about that. Fixed in 6848b7f8ccbbc371e066638e204f0be4c625194d

mmilch01 commented:
> > > Line 435: `$HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf=${MSMCONFIGDIR}/${MSMSulcConf} --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session"."$Hemisphere".sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/${Session}.${Hemisphere}.refsulc."$HighResMesh"k_fs_LR.shape.gii`
> > > Line 448: `$HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf=${MSMCONFIGDIR}/${MSMSulcConf} --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session"."$Hemisphere".sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/${Session}.${Hemisphere}.refsulc."$HighResMesh"k_fs_LR.shape.gii`
> > 
> > 
> > Addressed in [88c8080](https://github.com/Washington-University/HCPpipelines/commit/88c808014edfd06cf62fde7e7cd95a82b7e682d6)
> 
> The current commit doesn't have --refmesh/--refsulc, which default to the human templates in `HCPPIPEDIR`, not the copies in the subject. Matt's suggestion is to point `MSMSulc.sh` to the copies made earlier in the script.

Fixed in 6848b7f8ccbbc371e066638e204f0be4c625194d

glasserm commented:
Thanks @mmilch01 can you provide a test dataset via e-mail for review?

mmilch01 commented:
> Thanks @mmilch01 can you provide a test dataset via e-mail for review?

Yes, it was rerun on the same directory as before.

glasserm commented:
This is now ready to merge @coalsont.

coalsont commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -462,7 +432,7 @@ for Hemisphere in L R ; do
         if [ "$LongitudinalMode" == "NONE" ]; then
             cp "$AtlasSpaceFolder"/"$NativeFolder"/"$Session"."$Hemisphere".sphere.rot.native.surf.gii "$AtlasSpaceFolder"/"$NativeFolder"/MSMSulc/${Hemisphere}.sphere_rot.surf.gii
-            $HCPPIPEDIR/global/scripts/MSMSulc.sh --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere"
+            $HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf=${MSMCONFIGDIR}/${MSMSulcConf} --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session".HEMISPHERE.sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/${Session}.HEMISPHERE.refsulc."$HighResMesh"k_fs_LR.shape.gii
```
We added logic to `MSMSulc.sh` to handle configs outside of `MSMCONFIGDIR`, so we shouldn't be hardcoding that into the config path here (or below).

```suggestion
            $HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf="$MSMSulcConf" --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session".HEMISPHERE.sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/"$Session".HEMISPHERE.refsulc."$HighResMesh"k_fs_LR.shape.gii
```

coalsont commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -475,17 +445,16 @@ for Hemisphere in L R ; do
             #run MSMSulc.sh on average surface
-            $HCPPIPEDIR/global/scripts/MSMSulc.sh --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere"
-
+            $HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf=${MSMCONFIGDIR}/${MSMSulcConf} --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session".HEMISPHERE.sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/${Session}.HEMISPHERE.refsulc."$HighResMesh"k_fs_LR.shape.gii
```
```suggestion
            $HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf="$MSMSulcConf" --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session".HEMISPHERE.sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/"$Session".HEMISPHERE.refsulc."$HighResMesh"k_fs_LR.shape.gii
```

glasserm commented:
@mmilch01 Tim thinks the above needs testing to ensure no errors are introduced.

mmilch01 committed:
Apply suggestions from code review

Co-authored-by: Tim Coalson <coalsont@users.noreply.github.com>

coalsont commented on global/scripts/MSMSulc.sh:
```diff
@@ -38,6 +38,14 @@ SurfaceTemplateFolder="$HCPPIPEDIR"/global/templates/standard_mesh_atlases
+if [ ! -f "$ConfFile" ]; then 
+	log_Err_Abort "MSM config file $ConfFile does not exist"
+fi
+
```
There is a windows `CR` special character hiding in line 42, and there are trailing spaces after the `then`s.

coalsont commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -462,7 +432,7 @@ for Hemisphere in L R ; do
         if [ "$LongitudinalMode" == "NONE" ]; then
             cp "$AtlasSpaceFolder"/"$NativeFolder"/"$Session"."$Hemisphere".sphere.rot.native.surf.gii "$AtlasSpaceFolder"/"$NativeFolder"/MSMSulc/${Hemisphere}.sphere_rot.surf.gii
-            $HCPPIPEDIR/global/scripts/MSMSulc.sh --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere"
+            $HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf="$MSMSulcConf" --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session"."$Hemisphere".sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/"$Session"."$Hemisphere".refsulc."$HighResMesh"k_fs_LR.shape.gii
```
Why did you change these back to `"$Hemisphere"`?  MSMSulc.sh is expecting them to contain the literal string "HEMISPHERE" to replace with L/R depending on the `--hemi` argument.  At some point, I plan to make MSMSulc.sh error at least when `--hemi=B` and a literal "HEMISPHERE" isn't found in these arguments, to prevent right-to-left accidents.

Line 448 also.

coalsont committed:
fix stray CRLF, trailing whitespace and bracket consistency

coalsont commented:
I have rebased to pick up a more recent fix for MSMSulc.sh (mainly being able to display the help info without HCPPIPEDIR), minus the recent "$Hemisphere" commit, plus a fix for the CRLF.

@mmilch01, before doing anything else, please delete and re-create your local branch from the updated remote branch (**don't pull**, you can tag or make another branch name to preserve your old version first).  Something like `git remote update; git checkout --detach; git branch -d NHPUnifiedPostFreeSurfer; git checkout NHPUnifiedPostFreeSurfer`.

mmilch01 commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -462,7 +432,7 @@ for Hemisphere in L R ; do
         if [ "$LongitudinalMode" == "NONE" ]; then
             cp "$AtlasSpaceFolder"/"$NativeFolder"/"$Session"."$Hemisphere".sphere.rot.native.surf.gii "$AtlasSpaceFolder"/"$NativeFolder"/MSMSulc/${Hemisphere}.sphere_rot.surf.gii
-            $HCPPIPEDIR/global/scripts/MSMSulc.sh --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere"
+            $HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf="$MSMSulcConf" --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session".HEMISPHERE.sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/"$Session".HEMISPHERE.refsulc."$HighResMesh"k_fs_LR.shape.gii
```
```suggestion
            $HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf="$MSMSulcConf" --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session"."$Hemisphere".sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/"$Session"."$Hemisphere".refsulc."$HighResMesh"k_fs_LR.shape.gii
```

coalsont replied:
Please undo these.  If it causes an error, we should fix it in `MSMSulc.sh` so that it works as designed.

coalsont replied:
To be more specific, these are handled here:

https://github.com/Washington-University/HCPpipelines/blob/4c24a6e31ba2cc88ec6ec42f41b12c6c10c6c113/global/scripts/MSMSulc.sh#L85-L88

And since it did run (with the human templates) when we didn't specify refmesh/refsulc (and those defaults use a literal HEMISPHERE), I think it should already be working.

mmilch01 commented on PostFreeSurfer/scripts/FreeSurfer2CaretConvertAndRegisterNonlinear.sh:
```diff
@@ -475,17 +445,16 @@ for Hemisphere in L R ; do
             #run MSMSulc.sh on average surface
-            $HCPPIPEDIR/global/scripts/MSMSulc.sh --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere"
-
+            $HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf="$MSMSulcConf" --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session".HEMISPHERE.sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/"$Session".HEMISPHERE.refsulc."$HighResMesh"k_fs_LR.shape.gii
```
```suggestion
            $HCPPIPEDIR/global/scripts/MSMSulc.sh --msm-conf="$MSMSulcConf" --subject-dir="$StudyFolder" --subject="$Session" --regname="$RegName" --hemi "$Hemisphere" --refmesh="$AtlasSpaceFolder"/"$Session"."$Hemisphere".sphere."$HighResMesh"k_fs_LR.surf.gii --refdata="$AtlasSpaceFolder"/"$Session"."$Hemisphere".refsulc."$HighResMesh"k_fs_LR.shape.gii
```

mmilch01 committed:
Apply suggestions from code review

Re-apply HEMISPHERE literal fix

Co-authored-by: mmilch01 <mmilch01@gmail.com>

mmilch01 committed:
Revert HEMISPHERE literal fix

This reverts commit 4c24a6e31ba2cc88ec6ec42f41b12c6c10c6c113.

mmilch01 committed:
+fix accidental CRLF
